app/src/main/java/com/practicum/playlistmakerapp/media/ui/PlaylistsFragment.kt

```diff
@@ -22,11 +27,52 @@ class PlaylistsFragment : Fragment() {
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val adapter = PlaylistsAdapter(emptyList())
+        binding.recyclerView.adapter = adapter
+
+        binding.recyclerView.layoutManager = GridLayoutManager(requireContext(), 2)
+
+        binding.newPlaylistButton.setOnClickListener {
+            val intent = Intent(requireContext(), CreatePlaylistActivity::class.java)
+            startActivity(intent)
+        }
+
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.playlists.collect { playlists ->
+                if (playlists.isEmpty()) {
+                    showEmptyState()
+                } else {
+                    showPlaylists()
+                    adapter.playlists = playlists
+                    adapter.notifyDataSetChanged()
+                }
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        viewModel.loadPlaylists()
+    }
+
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
+    private fun showEmptyState() {
+        binding.recyclerView.visibility = View.GONE
+        binding.emptyStateLayout.visibility = View.VISIBLE
+    }
```

Comment:
```suggestion
    private fun showEmptyState() = with (binding) {
       recyclerView.visibility = View.GONE
       emptyStateLayout.visibility = View.VISIBLE
    }
```

можно через scope-функции, немного лаконичнее получается